Adds the `ey console` command to start a Rails console.

djhopper01 opened the pull request:
Hey guys,

First of all, thanks for this gem - I've used it a lot throughout the past few years.

This pull request is basically a helper/wrapper for this command.
`ey ssh -t 'cd /data/<app_name>/current && bundle exec rails console' -e <environment>`

I couldn't find any record of a command for this, so here's a pull request. It's very much similar to the ssh command (since it's an ssh session + running a command). This will end up being helpful to me even if this doesn't make it into the gem.

Before I finish this up and write the specs for it, I just want to see what you guys think about a command like this being in the gem. If you think it's worthy, I'll move it forward.

Re: code duplication, I see two alternatives.
1. Call the ssh command inside console command. But, I need to add the `shell` and `pty` options to the console command for this to work - it doesn't seem ideal since without `pty` it won't work at all.
2. Somehow pass options to the ssh command. The options are frozen, so that seems out of the question.

martinemde commented on lib/engineyard/cli.rb:
```diff
@@ -522,6 +522,47 @@ def ssh(cmd=nil)
+      desc: "Environment to console into"
+    method_option :account, type: :string, aliases: %w(-c),
+      required: true, default: '',
+      desc: "Name of the account in which the environment can be found"
```
Need to add the `method_option :app` also. You're using `fetch_app_environment` which is the correct fetcher, but it works better if it has the application name as an option (since, in particular, you want to open a specific application's rails console).

martinemde commented:
I'd be happy to merge and release a version with this command. If it gains enough traction we can expand it to cover consoles in other languages.

This will be especially useful if we ever make the thing I really want, which is to have the configuration script write out a generic `#{app_name}_console` bin script that can invoke a language specific console for any app. We have a quick hack to drop one like that one our main app, but we've never generalized it across languages.

I'd be happy to merge if you have it working. Check spelling everywhere (in particular can you fix the desc line at the top of the diff) and add a ChangeLog entry, with your name as committer too if you'd like.

djhopper01 commented:
Thanks @martinemde and no problem with the wait, I totally understand.

I've been using this myself for two weeks and it feels pretty good. I'll tidy this up a bit like you mentioned. If you don't mind, I'll also spend some time with the tests (since it looks like there's a spec for each command so far).

djhopper01 commented:
@martinemde I added a few specs that follow along with the specs for `ey ssh`. I added the ChangeLog entry, but I left the version and date blank for you. I'm assuming this would bump it up to 3.2.0.

Just let me know if you need anything else here. I hope people find it useful.

martinemde commented:
Looks great. Thank you!

martinemde commented:
Released [v3.2.0](https://github.com/engineyard/engineyard/releases/tag/v3.2.0).
